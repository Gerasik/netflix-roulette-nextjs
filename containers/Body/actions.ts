import ActionType from './constants';
import { ActionCreator } from './models';

export const setData: ActionCreator.SetData = payload => ({
  type: ActionType.SET_DATA,
  payload,
});

export const setStartData: ActionCreator.SetStartData = () => ({
  type: ActionType.SET_START_DATA,
});

// export const increment: Action. = payload => ({
//   type: ActionType.,
//   payload,
// });
