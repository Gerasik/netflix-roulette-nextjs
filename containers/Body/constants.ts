const ActionType = {
  SET_DATA: 'SET-DATA',
  SET_START_DATA: 'SET-START-DATA',
};

export default ActionType;
