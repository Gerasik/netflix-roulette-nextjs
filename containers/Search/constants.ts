const ActionType = {
  CHANGE_SEARCH_STRING: 'CHANGE-SEARCH-STRING',
  CHANGE_SEARCH_BY: 'CHANGE-SEARCH-BY',
  CHANGE_SORT_BY: 'CHANGE-SORT-BY',
};

export default ActionType;
