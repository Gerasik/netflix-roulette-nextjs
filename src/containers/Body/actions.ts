import ActionType from './constants';
import { ActionCreator } from './models';

export const setData: ActionCreator.SetData = payload => ({
  type: ActionType.SET_DATA,
  payload,
});

export const setStartData: ActionCreator.SetStartData = () => ({
  type: ActionType.SET_START_DATA,
});

export const addData: ActionCreator.AddData = () => ({
  type: ActionType.ADD_DATA,
});
