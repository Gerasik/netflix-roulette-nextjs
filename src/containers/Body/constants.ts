const ActionType = {
  SET_DATA: 'SET-DATA',
  SET_START_DATA: 'SET-START-DATA',
  ADD_DATA: 'ADD-DATA',
};

export default ActionType;
