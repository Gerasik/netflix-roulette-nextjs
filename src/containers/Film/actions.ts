import ActionType from './constants';
import { ActionCreator } from './models';

export const setFilmData: ActionCreator.SetFilmData = payload => ({
  type: ActionType.SET_FILM_DATA,
  payload,
});

export const setFilmId: ActionCreator.SetFilmId = payload => ({
  type: ActionType.SET_FILM_ID,
  payload,
});
