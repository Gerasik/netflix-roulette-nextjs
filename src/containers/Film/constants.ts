const ActionType = {
  SET_FILM_DATA: 'SET-FILM-DATA',
  SET_FILM_ID: 'SET-FILM-ID',
};

export default ActionType;
