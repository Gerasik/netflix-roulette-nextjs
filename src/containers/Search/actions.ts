import ActionType from './constants';
import { ActionCreator } from './models';

export const changeSearchString: ActionCreator.ChangeSearchString = payload => ({
  type: ActionType.CHANGE_SEARCH_STRING,
  payload,
});

export const changeSearchBy: ActionCreator.ChangeSearchBy = payload => ({
  type: ActionType.CHANGE_SEARCH_BY,
  payload,
});

export const changeSortBy: ActionCreator.ChangeSortBy = payload => ({
  type: ActionType.CHANGE_SORT_BY,
  payload,
});
