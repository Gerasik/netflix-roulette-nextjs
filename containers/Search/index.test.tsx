/* eslint-disable @typescript-eslint/camelcase */
import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { fromJS } from 'immutable';
import { mount } from 'enzyme';
import { REHYDRATE } from 'redux-persist/constants';

import { testSearchReducerData, testInitialState } from 'testData';
import searchReducer from './reducer';
import SearchContainer from '.';
import { changeSearchString, changeSearchBy, changeSortBy } from './actions';
import ActionType from './constants';

const middleware = [];
const mockStore = configureStore(middleware);

describe('Test dispatch action', () => {
  const store = mockStore(testInitialState);

  const testString = 'Hello';

  it('actions works right', () => {
    store.dispatch(changeSearchString(testString));
    store.dispatch(changeSearchBy(testString));
    store.dispatch(changeSortBy(testString));

    const actions = store.getActions();
    const expectedPayload = [
      { type: ActionType.CHANGE_SEARCH_STRING, payload: testString },
      { type: ActionType.CHANGE_SEARCH_BY, payload: testString },
      { type: ActionType.CHANGE_SORT_BY, payload: testString },
    ];

    expect(actions).toEqual(expectedPayload);
  });
});

describe('Test initial SearchContainer', () => {
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(testInitialState);
    wrapper = mount(
      <Provider store={store}>
        <SearchContainer />
      </Provider>
    );
  });

  it('should render with initial state data', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Test reducer', () => {
  it('should change search type', () => {
    const testData = 'hello';
    const reducerChange = searchReducer(fromJS(testSearchReducerData), changeSearchBy(testData));

    expect(reducerChange.get('searchBy')).toEqual(fromJS(testData));
  });

  it('should change sort type', () => {
    const testData = 'hello';
    const reducerChange = searchReducer(fromJS(testSearchReducerData), changeSortBy(testData));

    expect(reducerChange.get('sortBy')).toEqual(fromJS(testData));
  });

  it('should change search string', () => {
    const testData = 'hello';
    const reducerChange = searchReducer(
      fromJS(testSearchReducerData),
      changeSearchString(testData)
    );

    expect(reducerChange.get('searchString')).toEqual(fromJS(testData));
  });

  it('should load state from local storage', () => {
    const newPayload = fromJS({
      searchString: '',
      searchBy: 'rate',
      sortBy: 'date',
    });
    const reducerChange = searchReducer(fromJS(testSearchReducerData), {
      type: REHYDRATE,
      payload: {
        searchReducer: newPayload,
      },
    });
    expect(reducerChange).toEqual(newPayload);
  });

  it('should load initial state', () => {
    const reducerChange = searchReducer(fromJS(testSearchReducerData), {
      type: REHYDRATE,
      payload: {},
    });
    expect(reducerChange).toEqual(fromJS(testSearchReducerData));
  });

  it('should load initial state with unknown reducer', () => {
    const reducerChange = searchReducer(fromJS(testSearchReducerData), {
      type: 'unknown',
      payload: {},
    });
    expect(reducerChange).toEqual(fromJS(testSearchReducerData));
  });

  it('should load initial state with undefined state', () => {
    const reducerChange = searchReducer(undefined, {
      type: 'unknown',
      payload: {},
    });
    expect(reducerChange).toEqual(fromJS(testSearchReducerData));
  });
});
