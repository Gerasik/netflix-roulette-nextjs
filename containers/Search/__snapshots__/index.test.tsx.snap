// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test initial SearchContainer should render with initial state data 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Search)>
    <Search
      changeSearchBy={[Function]}
      changeSearchString={[Function]}
      searchData={
        Object {
          "searchBy": "genres",
          "searchString": "",
          "sortBy": "vote_average",
        }
      }
      setData={[Function]}
    >
      <p
        className="label"
      >
        find your movie
      </p>
      <div
        className="search-form"
      >
        <input
          className="search-text"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
        <button
          className="search-button"
          onClick={[Function]}
          type="button"
        >
          search
        </button>
      </div>
      <div
        className="search-by"
      >
        <span>
          search by
        </span>
        <button
          className="by-title"
          onClick={[Function]}
          type="button"
        >
          title
        </button>
        <button
          className="by-genre active"
          onClick={[Function]}
          type="button"
        >
          genre
        </button>
      </div>
    </Search>
  </Connect(Search)>
</Provider>
`;
